modin/core/storage_formats/base/query_compiler.py

```diff
@@ -430,6 +430,41 @@ def combine_first(self, other, **kwargs):  # noqa: PR02
     def eq(self, other, **kwargs):  # noqa: PR02
         return BinaryDefault.register(pandas.DataFrame.eq)(self, other=other, **kwargs)
 
+    def equals(
+        self,
+        other,
+        new_index: Optional[pandas.Index] = None,
+        new_columns: Optional[pandas.Index] = None,
+    ):
+        """
+        Test whether two objects contain the same elements.
+
+        If axes are not equal, perform frames alignment first. Note that
+        because frame alignment includes filling in empty values with NaN,
+        the query compiler for DataFrame([[1], [np.NaN]]) is equal to the query
+        compiler for DataFrame([[1]]]). In that case, the latter dataframe's
+        partition is expanded to DataFrame([[1], [np.NaN]]]).
+
+        Parameters
+        ----------
+        other : BaseQueryCompiler
+            The other compiler to be compared with the first.
```

Comment:
nit: is it more accurate to say that the other compiler is compared to self, as opposed to first?